outlinecaddy/ws2outline_handler.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2024 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package outlinecaddy
+
+import (
+	"errors"
+	"fmt"
+	"log/slog"
+	"net"
+	"net/http"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+	"github.com/Jigsaw-Code/outline-sdk/x/websocket"
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/modules/caddyhttp"
+	"github.com/mholt/caddy-l4/layer4"
+	"go.uber.org/zap"
+
+	onet "github.com/Jigsaw-Code/outline-ss-server/net"
+)
+
+const wsModuleName = "http.handlers.ws2outline"
```

Comment:
I wonder if we should use something more descriptive, like `websocket2conn` or `websocket2layer4`